src/Utilities/DoubleDicts.jl

```diff
@@ -94,7 +94,10 @@ mutable struct IndexDoubleDictInner{F,S} <:
                AbstractDoubleDictInner{F,S,MOI.ConstraintIndex{F,S}}
     dict::Dict{Int64,Int64}
     function IndexDoubleDictInner{F,S}(d::IndexDoubleDict) where {F,S}
-        return new{F,S}(get!(d.dict, (F, S), Dict{Int64,Int64}()))
+        if !haskey(d.dict, (F, S))
+            d.dict[(F, S)] = Dict{Int64,Int64}()
+        end
+        return new{F,S}(d.dict[(F, S)])
```

Comment:
```julia
        return new{F,S}(get!(d.dict, (F, S)) do
            Dict{Int64,Int64}()
        )
```

Reply:
this is possibly faster, because saves a haskey and does not create the dict if not needed.

Reply:
Yeah the problem previously was that it always created the dict. Your way needs an additional function compilation, which I guess is a trade-off against a key lookup? Anyway, it's a second-order consideration compared to creating a dict every time.